src/lib.rs

```diff
@@ -834,10 +841,11 @@ pub fn max_level() -> LevelFilter {
 ///
 /// Requires the `use_std` feature (enabled by default).
 #[cfg(feature = "use_std")]
-pub fn set_logger<M>(make_logger: M) -> Result<(), SetLoggerError>
-    where M: FnOnce(MaxLevelFilter) -> Box<Log>
-{
-    unsafe { set_logger_raw(|max_level| mem::transmute(make_logger(max_level))) }
+pub fn set_logger(logger: Box<Log>, filter: LevelFilter) {
+    match try_set_logger(|max| { max.set(filter); logger }) {
+        Ok(()) => {}
+        Err(_) => panic!("global logger is already initialized"),
+    }
 }
```

Comment:
The docs for this function will also need a "# Panics" section to describe the conditions when this function panics.